homeassistant/components/linky/sensor.py

```diff
@@ -68,21 +124,33 @@ def unit_of_measurement(self):
         """Return the unit of measurement."""
         return ENERGY_KILO_WATT_HOUR
 
-    @Throttle(SCAN_INTERVAL)
-    def update(self):
-        """Fetch new state data for the sensor."""
-        from pylinky.client import PyLinkyError
-        try:
-            self._client.fetch_data()
-        except PyLinkyError as exp:
-            _LOGGER.error(exp)
-            self._client.close_session()
-            return
+    @property
+    def icon(self):
+        """Return the icon of the sensor."""
+        return ICON_ENERGY
 
-        _LOGGER.debug(json.dumps(self._client.get_data(), indent=2))
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the sensor."""
+        if self._client.get_data()[self._scale][self._when][TIME]:
+            time = self._client.get_data()[self._scale][self._when][TIME]
+            if self._scale is not YEARLY:
+                year_index = INDEX_CURRENT
+                if self._client.get_data()[MONTHLY][self._when][TIME] == "Dec":
+                    year_index = INDEX_LAST
+                time += ' ' + self._client.get_data()[YEARLY][year_index][TIME]
+            return {
+                ATTR_ATTRIBUTION: ATTRIBUTION,
+                'last_updated': self._last_updated,
+                'time': time
+            }
+        return None
 
-        if self._client.get_data():
-            # get the last past day data
-            self._state = self._client.get_data()['daily'][-2]['conso']
+    def update(self):
+        """Fetch new state data for the sensor."""
+        if self._client.get_data()[self._scale][self._when][CONSUMPTION]:
+            self._state = self._client.get_data()[self._scale][
+                self._when][CONSUMPTION]
+            self._last_updated = datetime.now().isoformat()
```

Comment:
Rewrite that to make only one call aka store the result in a variable. E.g., 

```python
    self.data = self._client.get_data()[self._scale][self._when]
    self._state = self.data[CONSUMPTION]
```

I assume that there will be a traceback if there are no data, the data can be retrieved or the result is not complete.

Reply:
Yep, there is here https://github.com/home-assistant/home-assistant/pull/23726/files#diff-ac8d0d74d5aabe103c0e92d309048995R71

<img width="1282" alt="Capture d’écran 2019-05-08 à 19 53 08" src="https://user-images.githubusercontent.com/14821482/57397858-e5413100-71cd-11e9-9e4f-0ab647dc2ec0.png">